embeddings.mdx

```diff
@@ -0,0 +1,151 @@
+    string type. If you would like to use multiple features as input, you can define a [resolver](/docs/resolver) to map
+    these features into one that will be used as the input.
+1.  `provider` (required): The embedding model provider. The currently supported providers are `sentence-transformers`, `instructor`,
+    `openai`, or `cohere`. Chalk may add more providers in the future.
+1.  `model` (optional): The name of the model to use. Each model family has a default model, which will be used if this
+    argument is not specified.
+1.  `ttl` (optional): The duration for which the embedding will be cache. By default, the embedding vector will be
+    cached for the same TTL as the `feature`. If you would like different behavior,
+    you can specify this argument explicitly.
+
+For the complete signature, please see the [api docs](/api-docs#embedding).
+
+
+
+### Sentence Transformers
+
+Chalk supports all models that are part of the [sentence-transformers](https://www.sbert.net/) framework.
+The default model for this provider is `all-MiniLM-L6-v`. The full list of models is available
+[here](https://www.sbert.net/docs/pretrained_models.html).
+
+```python
+from chalk.features import embedding, features, Vector
+
+@features
+class Document:
+    content: str
+    embedding: Vector = embedding(
+        feature="content",
+        provider="sentence-transformers",
+        model="all-MiniLM-L6-v",  # optional, defaults to all-MiniLM-L6-v
+    )
+```
+
+### Instructor
+
+Chalk supports [INSTRUCTOR](https://github.com/xlang-ai/instructor-embedding) embedding models.
+The default model for this provider is `hkunlp/instructor-base`. The full list of models is available
+[here](https://github.com/xlang-ai/instructor-embedding#model-list).
+
+```python
+from chalk.features import embedding, features, Vector
+
+@features
+class Document:
+    content: str
+    embedding: Vector = embedding(
+        feature="content",
+        provider="instructor",
+        model="hkunlp/instructor-base",  # optional, defaults to hkunlp/instructor-base
+    )
+```
+
+### OpenAI
+
+Chalk can proxy calls to the [OpenAI Embeddings API](https://platform.openai.com/docs/guides/embeddings).
+The default model for this provider is `text-embedding-ada-002`.
+To use OpenAI, first sign up for an
+[OpenAI Account](https://openai.com/api/), and then create an [OpenAI Integration](/docs/openai) in Chalk.
+All OpenAI requests will be attributed to your API key. To minimize usage, we highly recommend specifying an
+appropriate TTL in Chalk, which will ensure that embeddings are cached.
+
+```python
+from chalk.features import embedding, features, Vector
+
+@features
+class Document:
+    content: str
+    embedding: Vector = embedding(
+        feature="content",
+        provider="openai",
+        model="text-embedding-ada-002",  # optional, defaults to text-embedding-ada-002
+        ttl="infinity",
```

Comment:
is this really ttl or is it max_staleness? we have `offline_ttl` somewhere, and it means we'll delete it after that time

Reply:
Right, `max_staleness`